.github/workflows/zebra-test-ci.yaml

```diff
@@ -0,0 +1,79 @@
+# CI Workflow to test tx-tool against a locally built Zebra node (defined in "Dockerfile-demo")
+# This workflow ensures tx-tool works correctly by spinning up a Zebra container
+# and running the tx-tool from the current branch against it in a Docker network.
+# Triggers: push to main, pull requests, and manual runs via workflow_dispatch.
+
+name: Check tx-tool against Zebra
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+      
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+      
+    - name: Build Zebra Node (Dockerfile-demo)
+      uses: docker/build-push-action@v4
+      with:
+        context: .
+        file: ./Dockerfile-demo
+        tags: zebra-node:custom
+        push: false
+        load: true
+        
+    - name: Run Zebra Node container
+      run: |
+        docker run -d --name zebra-node -p 18232:18232 zebra-node:custom
+        # Wait for the node to start up
+        echo "Waiting for Zebra node to initialize..."
+        sleep 30
+        echo "Finished waiting for Zebra node..."
+        
+    - name: Build tx-tool
+      uses: docker/build-push-action@v4
+      with:
+        context: .
+        file: ./Dockerfile
+        tags: zcash-tx-tool:custom
+        push: false
+        load: true
+        
+    - name: Run tx-tool and verify success
+      run: |
+        # Create a network to connect the containers
+        docker network create zcash-net
+        
+        # Connect zebra-node to the network
+        docker network connect zcash-net zebra-node
+        
+        # Run the tx tool and capture exit code
+        docker run --network zcash-net \
+          -e ZCASH_NODE_ADDRESS=zebra-node \
+          -e ZCASH_NODE_PORT=18232 \
+          -e ZCASH_NODE_PROTOCOL=http \
+          --name zcash-tx-tool \
+          zcash-tx-tool:latest
```

Comment:
what guarentees that this is the latest changes to the tx-tool code? (where zcash-tx-tool:latest is published) ?

Reply:
i see `zcash-tx-tool:custom` above, not `:latest`